### PR TITLE
support quant_lm_head arg in all WOQ configs

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

feature

## Description

- [x] add quant_lm_head arg in WOQ configs
- [x] enhance UTs to test it
- [x] enhance GPTQ calibration forward wrapper to avoid explicit ValueError
- [x] RTN, AWQ, TEQ, and HQQ can quantize lm_head, GPTQ is not yet supported.

## Expected Behavior & Potential Risk

UTs pass and WOQ model test pass

## How has this PR been tested?

local test

### CODE REVIEW

File: neural_compressor/common/base_config.py [hunk 1]
```diff
@@ -198,10 +198,25 @@ def local_config(self):
     def local_config(self, config):
         self._local_config = config
 
-    def set_local(self, operator_name: Union[str, Callable], config: BaseConfig) -> BaseConfig:
-        if operator_name in self.local_config:
-            logger.warning("The configuration for %s has already been set, update it.", operator_name)
-        self.local_config[operator_name] = config
+    def set_local(self, operator_name_or_list: Union[List, str, Callable], config: BaseConfig) -> BaseConfig:
+        """Set custom configuration based on the global configuration object.
```
Comment (yiliu30) [inline neural_compressor/common/base_config.py:202]: based on the local configuration object?

File: neural_compressor/torch/quantization/config.py [hunk 2]
```diff
@@ -104,6 +104,8 @@ class RTNConfig(BaseConfig):
         "double_quant_bits",
         "double_quant_use_sym",
         "double_quant_group_size",
+        # quant_lm_head
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:107]:
```suggestion
```

File: neural_compressor/torch/quantization/config.py [hunk 3]
```diff
@@ -467,10 +505,19 @@ def register_supported_configs(cls) -> List[OperatorConfig]:
...
-        operators = [torch.nn.Linear, torch.nn.functional.linear]
+        operators = list(WOQ_WHITE_LIST)
         supported_configs.append(OperatorConfig(config=linear_awq_config, operators=operators))
         cls.supported_configs = supported_configs
 
+    def to_config_mapping(
+        self, config_list: List[BaseConfig] = None, model_info: List[Tuple[str, str]] = None
+    ) -> OrderedDictType[Union[str, str], OrderedDictType[str, BaseConfig]]:
+        if not self.quant_lm_head:
+            usual_lm_head_names = [".*lm_head", ".*output_layer", ".*embed_out"]
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:516]: Move it to `constants.py`, if possible.

**Review by yiliu30 — APPROVED** [review]

**Review by Kaihui-intel — APPROVED** [review]

File: neural_compressor/common/base_config.py (hunk 1 above)
Comment (xin3he) [reply neural_compressor/common/base_config.py:202]: I think that the config initialized by user is global config, global for all ops.

**Pull request merged**